### PR TITLE
Feature/retry assets

### PR DESCRIPTION
**Pull request by angarita-dev**
## Changelog

- Allows for refresh if token balance is `Error`
- Allows manual refresh on failed balance
- Changed `getBalance` method on `Background/Keyring` to match with new `getBalances`

### Type of changes included:

- [x] Components
- [x] Business Logic
- [ ] Bug fixes
- [ ] Styling
- [x] Code Refactor

### Clubhouse Tickets Related:

- [sc-24817](https://app.shortcut.com/terminalsystems/story/24817/feature-manual-token-balance-refresh)

### Screenshots/Gifs:

*Place you changes screenshots here*

### Notes:

*Place special notes here*

### Tested on:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Review by mattgle — APPROVED** [review]
Looks good but remember that i don't know Plug Extension that well

File: source/shared/constants/currencies.js [hunk 1]
```diff
@@ -44,29 +44,30 @@ export const currencyPropTypes = {
 export const CYCLES_PER_TC = 1_000_000_000_000;
 
 export const formatAssetBySymbol = (_amount, symbol, icpPrice) => {
-  const amount = parseFloat(_amount, 10);
+  const error = _amount === 'Error';
+  const amount = error ? 'Error' : parseFloat(_amount, 10);
```
Comment (rocky-fleek) [inline source/shared/constants/currencies.js:48]: `error || parseFloat(_amount, 10)`

File: source/components/Tokens/index.jsx [hunk 2]
```diff
@@ -43,7 +44,13 @@ const Tokens = () => {
       <div className={classes.tokenContainer}>
         {
           assets?.map((asset) => (
-            <AssetItem {...asset} key={asset.name} loading={loading} />
+            <AssetItem
+              {...asset}
+              updateToken={fetchAssets}
+              key={asset.name}
+              loading={loading}
+              failed={asset.amount === 'Error'}
```
Comment (rocky-fleek) [inline source/components/Tokens/index.jsx:52]: we should save 'Error' in a constant to be easier to mantain

File: source/shared/constants/currencies.js [hunk 3]
```diff
@@ -44,29 +44,30 @@ export const currencyPropTypes = {
...
   return (
     {
       ICP: {
         amount,
-        value: amount * icpPrice,
+        value: error ? 'Error' : amount * icpPrice,
```
Comment (rocky-fleek) [inline source/shared/constants/currencies.js:53]: same as before (for all of this file)

**Pull request merged**